### PR TITLE
Fix is_focus_priority_greater function

### PR DESCRIPTION
**Pull request by emersion**
Supersedes https://github.com/Plagman/gamescope/pull/173

### CODE REVIEW

File: src/steamcompmgr.cpp [hunk 1]
```diff
@@ -1318,29 +1324,40 @@ win_skip_taskbar_and_pager( win *w )
...
-		return false;
+	if ( win_skip_taskbar_and_pager( a ) != win_skip_taskbar_and_pager ( b ) )
+		return !win_skip_taskbar_and_pager( a );
 
-	if ( a->damage_sequence < b->damage_sequence )
-		return false;
+	if ( a->damage_sequence != b->damage_sequence )
+		return a->damage_sequence > b->damage_sequence;
 
 	return true;
```
Comment (lostgoat) [inline src/steamcompmgr.cpp:1362]: This should now be a `return false`

File: src/steamcompmgr.cpp [hunk 2]
```diff
@@ -1318,29 +1324,40 @@ win_skip_taskbar_and_pager( win *w )
...
 	// Wine sets SKIP_TASKBAR and SKIP_PAGER hints for WS_EX_NOACTIVATE windows.
 	// See https://github.com/Plagman/gamescope/issues/87
-	if ( win_skip_taskbar_and_pager( a ) && !win_skip_taskbar_and_pager ( b ) )
-		return false;
+	if ( win_skip_taskbar_and_pager( a ) != win_skip_taskbar_and_pager ( b ) )
+		return !win_skip_taskbar_and_pager( a );
 
-	if ( a->damage_sequence < b->damage_sequence )
-		return false;
+	if ( a->damage_sequence != b->damage_sequence )
```
Comment (lostgoat) [inline src/steamcompmgr.cpp:1359]: This is missing an equivalent to https://github.com/Plagman/gamescope/pull/173/commits/e19cd012037fb8f486c2cc8df5e2d2d482cd7a61

**Review by lostgoat — CHANGES_REQUESTED** [review]
Hi Simon!

Gave this a try and ran into some issues with steam popups. Left some comments with required changes.

**Comment by emersion** [comment]
Thanks, these are good points! Updated the PR.

**Comment by lostgoat** [comment]
Tests okay here :+1:

**Comment by lostgoat** [comment]
Thanks for merging!

**Pull request merged**